src/chains/ethereum/ethereum/src/api.ts

```diff
@@ -2884,6 +2884,206 @@ export default class EthereumApi implements Api {
+   * block is empty.
+   *
+   * @example
+   * ```javascript
+   * const [from, to] = await provider.request({ method: "eth_accounts", params: [] });
+   * await provider.request({ method: "eth_sendTransaction", params: [{ from, to }] });
+   * const feeHistory = await provider.request({ method: "eth_feeHistory", params: ["0x1", "0x1", [10, 100]] });
+   * console.log(feeHistory);
+   * ```
+   */
+  @assertArgLength(3)
+  async eth_feeHistory(
+    blockCount: QUANTITY,
+    newestBlock: QUANTITY | Ethereum.Tag,
+    rewardPercentiles: number[]
+  ): Promise<Ethereum.FeeHistory<"private">> {
+    const blockchain = this.#blockchain;
+    const MIN_BLOCKS: number = 1;
+    const MAX_BLOCKS: number = 1024;
+    const PRECISION_FLOAT: number = 1e14;
+    const PAD_PRECISION = 16;
+    const PRECISION_BIG_INT: bigint = BigInt(1e16);
+
+    const newestBlockNumber = blockchain.blocks
+      .getEffectiveNumber(newestBlock)
+      .toBigInt();
+
+    // blockCount must be within MIN_BLOCKS and MAX_BLOCKS. blockCount > newestBlock is
+    // technically valid per the spec but we cannot go past the Genesis Block. Values
+    // above MAX_BLOCKS are technically within spec, however we cap totalBlocks because
+    // of the resource needs and potential abuse of a very large blockCount.
+    const totalBlocks = Math.min(
+      Math.max(
+        Quantity.toBigInt(blockCount) >= MAX_BLOCKS
+          ? MAX_BLOCKS
+          : Quantity.toNumber(blockCount),
+        MIN_BLOCKS
+      ),
+      Quantity.toNumber(
+        newestBlockNumber >= MAX_BLOCKS ? MAX_BLOCKS : newestBlockNumber + 1n
+      ),
+      MAX_BLOCKS
+    );
+
+    const baseFeePerGas: Quantity[] = new Array(totalBlocks);
+    const gasUsedRatio: number[] = new Array(totalBlocks);
+    let reward: Array<Quantity[]>;
+
+    // percentiles must be unique and in ascending order between 0 and 100
+    if (rewardPercentiles.length > 0) {
+      const ERR_INVALID_PERCENTILE =
+        "invalid reward percentile: percentiles must be unique and in ascending order";
+      if (rewardPercentiles[0] < 0)
+        throw new Error(`${ERR_INVALID_PERCENTILE} ${rewardPercentiles[0]}`);
+      if (rewardPercentiles[rewardPercentiles.length - 1] > 100)
+        throw new Error(
+          `${ERR_INVALID_PERCENTILE} ${
+            rewardPercentiles[rewardPercentiles.length - 1]
+          }`
+        );
+
+      for (let i = 1; i < rewardPercentiles.length; i++) {
+        if (rewardPercentiles[i] <= rewardPercentiles[i - 1]) {
+          throw new Error(
+            `${ERR_INVALID_PERCENTILE} ${rewardPercentiles[i]} ${
+              rewardPercentiles[i - 1]
+            }`
+          );
+        }
+      }
+
+      reward = new Array(totalBlocks);
+    }
+    // totalBlocks is inclusive of newestBlock
+    const oldestBlockNumber =
+      newestBlockNumber - Quantity.from(totalBlocks - 1).toBigInt();
```

Comment:
```suggestion
      newestBlockNumber - BigInt(totalBlocks - 1);
```
This will avoid some intermediary conversions